assets/terraform-modules/tinkerbell-sandbox/worker/main.tf

```diff
@@ -0,0 +1,64 @@
+resource "libvirt_volume" "worker" {
+  name = var.name
+  pool = var.sandbox.volumes_pool_name
+  size = 20 * 1000 * 1000 * 1000 # 20GB
+}
+
+# Generate random numbers in hex format to generate worker MAC address.
+resource "random_id" "mac_address" {
+  count       = 5
+  byte_length = 1
+}
+
+locals {
+  # 52 is a Locally Administered Address Range, so it's safe to use.
+  mac = "52:${join(":", random_id.mac_address.*.hex)}"
+}
+
+resource "libvirt_domain" "worker" {
+  name   = "${var.sandbox.sandbox_name}-${var.name}"
+  vcpu   = 4
+  memory = 4096
```

Comment:
Hm, it will be complicated, as we don't expose any fields for this module in HCL, and having another block for configuring libvirt worker pool seems like an overkill to me.

4GB is a reasonable value for both controller node and worker node, as controller must be able to run 2 kube-apiserver processes at a time (1-1,5GB each), 300MB for etcd, 500MB for OS, 100MB for kubelet, 100MB for kube-proxy.

Any ideas how to make it configurable in non-overly complex way?

Reply:
Don't have ideas now but with the provisioner's RAM reduced I'm happy with the memory requirements now.